src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -248,6 +248,17 @@ private static ImmutableArray<ValidationError> Validate(BenchmarkRunInfo[] bench
 
             var validationErrors = new List<ValidationError>();
 
+            if (benchmarks.Any(b => b.Config.Options.IsSet(ConfigOptions.JoinSummary)))
+            {
+                var joinedCases = benchmarks.SelectMany(b => b.BenchmarksCases).ToList();
```

Comment:
nit: if you don't plan to add any items to the collection, an array is a better choice than list (and here `ValidationParameters` accepts an `IReadOnlyList`)

```suggestion
                var joinedCases = benchmarks.SelectMany(b => b.BenchmarksCases).ToArray();
```